### ISSUE TITLE
Issues on M1 MBP

### ISSUE THREAD

**Issue by ghost** [bryce-co/XcodeVimMap#1]
I'm trying to get this working on a M1 MBP running Xcode-13.0b4

I have had no success getting this to work unless I build using `My Mac (rosetta)` then all is good when Xcode launches it works as expected, it isn't until I close Xcode and reopen it the issues start, when running the terminal command to see if the plugin was loaded I am greeted with this error:

```
2021-08-27 18:58:10.404 Xcode[44837:2917834] WARNING: Failed to load plugin at path: "/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin", skipping. Error: Error Domain=NSCocoaErrorDomain Code=3585 "dlopen_preflight(/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap): no suitable image found.  Did find:
	/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap: mach-o, but wrong architecture
	/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap: mach-o, but wrong architecture" UserInfo={NSLocalizedFailureReason=The bundle doesn’t contain a version for the current architecture., NSLocalizedRecoverySuggestion=Try installing a universal version of the bundle., NSFilePath=/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap, NSDebugDescription=dlopen_preflight(/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap): no suitable image found.  Did find:
	/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap: mach-o, but wrong architecture
	/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin/Contents/MacOS/XcodeVimMap: mach-o, but wrong architecture, NSBundlePath=/Users/carlos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeVimMap.xcplugin, NSLocalizedDescription=The bundle “XcodeVimMap” couldn’t be loaded because it doesn’t contain a version for the current architecture.}
```

I tried checking the box to run Xcode under rosetta to no avail

Any help getting this amazing plugin working would be greatly appreciated.

**Comment by meieri** [comment]
I got this working on an M1 MB air running Xcode Version 13.2.1 (13C100). 

First of all, thank you very much Bryce for your fantastic blog post. I have the exact same problem, and binged your post at 4am last night. Super fascinating and well done. 

I read this issue before I tried to get this working, and my solution was to just build the plugin myself. I followed Bryce's clearly outlined steps in the blog post to setup the plugin, although note that at time of writing instead of installing the plugin to ```~/Library/Application\ Support/Developer/Shared/Xcode/Plug-ins/``` (which no longer exists)
I had to drop the plugin into ```/Applications/Xcode.app/Contents/PlugIns```. 

Couldn't find on google, so for anyone else looking: Plugin install directory new Xcode xcode 13 

Unfortunately they changed some stuff that makes checking for insert mode crash. Honestly, I just want the functionality, so I disabled that code altogether. It was quite crazy if you read the post. Very cool but I care way less about sending an erroneous esc char in normal mode then i do having this functionality. 

With these changes, it works, and it's lifechanging. Thank you kindly for your work Bryce and hope this helps someone.

**Comment by ghost** [comment]
Hi @meieri Thank you very much i'll give it another go as I deeply miss having this functionality.

Could you please tell me which part of the code you disabled.

Thanks again

**Comment by kkharji** [comment]
fascinating article @brycepauken, thanks so much 


So after sometime writing the code and trying to shy away from lldb :D 

I endup with a crash right after calling `object_getIvar`.

```yml
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000020
Exception Codes:       0x0000000000000001, 0x0000000000000020
Exception Note:        EXC_CORPSE_NOTIFY


Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [50885]
```

So after trying to get to the point where it faulted,  it turn out `getIvar:@"mode" from:vimContext` is where it faulted. Maybe I'd try lldb one more time and check the content of vimContext or where mode is defined :D 

btw @meieri thank you for `/Applications/Xcode.app/Contents/PlugIns` while the old path worked fine, it definitely saved from the annoying errors of mismatch id when using apple singed Xcode. btw using this way, did you have to sign your own Xcode version?

edit: vimContext indeed has mode 🤔 

```swift
ViEventConsumer.Context<mode: .normal, keyStroke: ViKeystroke(keyEquivalents: []), currentKeyEquivalentIndex: 0, register: nil, count: nil, partialCommands: [], operatorCommand: nil, lastMotionCommand: Optional(SourceEditor.ViMoveParagraphDownCommand), lastMotionCommandRanges = Optional([Range(SourceEditorPosition: line: 102, col: 0..<SourceEditorPosition: line: 104, col: 0)])>
```

**Comment by brycebostwick** [comment]
I'm not on github often so I'm super behind  on this - short version, @kukushi  kindly fixed this in #3 , possibly with the help of the discussion in this thread, so thanks all ❤️
